lib/misty/http/method_builder.rb

```diff
@@ -86,10 +86,16 @@ def prefixed_path(path)
 
       def query_param(data)
         if data.is_a? String
+          return '' if data == ''
           str = '?'
           str.force_encoding('ASCII-8BIT')
-          str << data unless data.empty?
+          str << data
+          return str
+        elsif data.is_a? Hash
+          return '' if data.empty?
+          return '?' + URI.encode_www_form(data)
         end
+        return nil
```

Comment:
Uhmmm... okay. I'm a friend of exiting early because code flow can easily get confusing otherwise. But if you wish, I'll rewrite this.